File: src/main/java/com/example/springsecurity/entity/Technician.java
package com.example.springsecurity.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Technician{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String username;
    private String password;
    private String emailAddress;
    private String firstName;
    private String lastName;
    private Date dob;
    private String phoneNumber;
    private String city;
    private String roles;
    private String domain;
    private Date startDate;
}

File: el_ousta/lib/models/Technician.dart
class Technician {
  final String username;
  final String password;
  final String emailAddress;
  final String firstName;
  final String lastName;
  final DateTime dob;
  final String phoneNumber;
  final String roles;
  final String city;
  final String domain;
  final DateTime startDate;

  Technician(
      {
        required this.username,
        required this.password,
        required this.emailAddress,
        required this.firstName,
        required this.lastName,
        required this.dob,
        required this.phoneNumber,
        required this.city,
        required this.roles,
        required this.domain,
        required this.startDate
      }
    );
  Map<String, dynamic> toJson() {
    return {
      'username': username,
      'password': password,
      'emailAddress': emailAddress,
      'firstName': firstName,
      'lastName': lastName,
      'dob': dob.toIso8601String(), // Convert DateTime to ISO format
      'phoneNumber': phoneNumber,
      'city': city,
      'roles': roles,
      'domain': domain,
      'startDate': startDate.toIso8601String(), // Convert DateTime to ISO format
    };
  }
}